UnityProject/Assets/Scripts/Objects/Drawers/Cremator.cs

```diff
@@ -107,55 +121,65 @@ public override void CloseDrawer()
 			UpdateCloseState();
 		}
 
+		public override void OpenDrawer()
+		{
+			base.OpenDrawer();
+			if(drawerState == (DrawerState)CrematorState.ShutAndActive) StopCoroutine(BurnContent());
+		}
+
 		private void UpdateCloseState()
 		{
 			if (container.IsEmpty == false)
 			{
 				SetDrawerState((DrawerState)CrematorState.ShutWithContents);
+				return;
 			}
-			else SetDrawerState(DrawerState.Shut);
+			SetDrawerState(DrawerState.Shut);
 		}
 
 		private void Cremate()
 		{
-			OnStartPlayerCremation();
-			StartCoroutine(PlayIncineratingAnim());
 			SoundManager.PlayNetworkedAtPos(CremationSound, DrawerWorldPosition, sourceObj: gameObject);
+			SetDrawerState((DrawerState)CrematorState.ShutAndActive);
+			UpdateCloseState();
+			OnStartPlayerCremation();
+			StartCoroutine(nameof(BurnContent));
 		}
 
-		private void DestroyContents()
+		private IEnumerator BurnContent()
 		{
 			foreach (var obj in container.GetStoredObjects())
 			{
-				if (obj.TryGetComponent<PlayerScript>(out var script) && script.mind != null)
-				{
-					PlayerSpawn.ServerSpawnGhost(script.mind);
-				}
-
-				_ = Despawn.ServerSingle(obj);
-				container.RemoveObject(obj);
+				if(obj.TryGetComponent<Integrity>(out var integrity)) //For items
+					integrity.ApplyDamage(burningDamage, AttackType.Fire, DamageType.Burn, true);
+				if (obj.TryGetComponent<LivingHealthBehaviour>(out var healthBehaviour)) //For NPCs
+					healthBehaviour.ApplyDamage(gameObject, burningDamage, AttackType.Fire, DamageType.Burn);
+				if (obj.TryGetComponent<PlayerHealthV2>(out var playerHealthV2)) //For Players
+					playerHealthV2.ApplyDamageAll(gameObject, burningDamage, AttackType.Fire, DamageType.Burn);
 			}
+
+			yield return WaitFor.Seconds(BURNING_DURATION);
+			StartCoroutine(nameof(BurnContent));
```

Comment:
not really about code but, should we have these things that can remove a player from a round Permanently?

Reply:
I removed the insta player removal because thats not how /tg/ handles it, the player's body needs to burn for a while until it ashes so they can be removed from the game entirely. Doing it this way allows players to have time to rescue antag victims who were attempting to destroy evidence.